core/src/local_cluster.rs

```diff
@@ -198,30 +208,45 @@ impl LocalCluster {
         }
     }
 
-    fn add_validator(&mut self, fullnode_config: &FullnodeConfig, stake: u64) {
+    fn add_validator(&mut self, mut fullnode_config: FullnodeConfig, stake: u64) {
```

Comment:
Requiring a mutable `fullnode_config` feels a little rude.  "Why does adding a validator require modifying the configuration I want to give it"?  Can we push that `fullnode_config.voting_disabled = true` to the caller?

Reply:
It doesn't. The rude part is requiring the FullnodeConfig be moved instead of borrowed. That's just a Rust idiom that'd otherwise read:

```rust
fn foo(bar: Bar) {
    let mut bar = bar;
    ...
}
```